### PR TITLE
COMPASS-3971 select favorite after disconnect fix

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Show connection details of the current favorite after disconnecting
- If a new created favorite does not have `replicaSet` or `mongodbDatabaseName` values, then when switching to another favorite that has these values and switching back should properly set or unset these values in form inputs.
- Fix responsiveness problem on the `More Options` tab. When selecting SLL values that contain buttons, the button should remain within the form.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by alenakhineika** [comment]
@lrlna I have updated this PR after your approval to make the implementation more universal. Before I was unsetting just some form values implicitly to fix a bug. But with the new version, all properties even not visible ones are being unset to prevent future issues. Pls take a look again and let me know if I can merge.

File: src/stores/index.js [hunk 1]
```diff
@@ -796,15 +793,9 @@ const Store = Reflux.createStore({
...
-    connection.replicaSet = connection.replicaSet || '';
-    connection.mongodbDatabaseName = connection.mongodbDatabaseName || '';
-
     this.state.currentConnection = connection;
     this.state.fetchedConnections.add(connection);
-    this.state.connections[connection._id] = connection.getAttributes({
-      props: true,
-      derived: true
-    });
+    this.state.connections[connection._id] = { ...connection._values };
```
Comment (lrlna) [inline src/stores/index.js]: so the `connection._values` is going to have the correct `.replicaSet` and have it set to be an empty string if none?

Comment (alenakhineika) [reply src/stores/index.js]: @lrlna yes. `getAttributes` method returns only properties that have some value but `_values` attribute stores all properties including those that have `undefined` value. Therefore when we save a favorite we save all properties, and when we switch between favorites we update all current connection properties. In the case of `undefined` value, we unset a property.

Comment (lrlna) [reply src/stores/index.js]: ahhhh! gotcha! this is way better than doing it manually then!

**Pull request merged**